thunderfish/eodanalysis.py

```diff
@@ -0,0 +1,174 @@
+    :return freqs:          (1-D array) frequency array of a psd.
+    """
+    cfg = ct.get_config_dict()
+
+    if verbose is not None:
+        cfg['verboseLevel'][0] = verbose
+    channel = channel
+
+    # load data using dataloader module
+    print('loading example data ...\n')
+    data, samplrate, unit = dl.load_data(audio_file)
+
+    # calculate best_window
+    print('calculating best window ...\n')
+    bwin_data, clip = bw.best_window(data, samplrate)
+
+    print('calculating powerspectrum ...\n')
+    power, freqs = ps.powerspectrum(data, samplrate)
+
+
+    psd_type, proportion = pt.psd_assignment(power, freqs)
+
+    return bwin_data, samplrate, psd_type
+
+def eod_analysis(bwin_data, samplerate, fish_type, psd_type, plot_data_func=None, **kwargs):
+    """
+    Detects EODs in the given data and tries to build a mean eod.
+
+    This function takes the data array of a soundfile, the samplerate and several other arguments to detect EODs and
+    build a mean eod with the help of several other functions.
+
+    :param bwin_data:           (1-D array) containing the data that shall be analysed.
+    :param samplerate:          (float) samplerate of the data that shall be analysed.
+    :param fish_type:           (string) result of the "sortfishtype.py" modul.
+    :param psd_type:            (string) result of the "psdtype.py" mudule.
+    :param plot_data_func:      (function) function (eodanalysisplot()) that is used to create a axis for later plotting containing a figure to
+                                visualice what the modul did.
+    :param kwargs:              additional arguments that are passed to the plot_data_func().
+    :return mean_eod:           (1-D array) containing the data of the mean eod.
+    :return std_eod:            (1-D array) containing the data of the std eod.
+    :return time:               (1-D array) containing the time of the mean eod. (same length as mean_eod or std_eod)
+    :return ax:                 (axis for plot) axis that is ready for plotting explaining what the modul does.
+    """
+
+    print('\nAnalysing EOD structures ...')
+    pulse_data = eod_extracting(bwin_data, samplerate, fish_type, psd_type)
+    mean_eod, std_eod = eod_mean(pulse_data)
+
+    time = ((np.arange(len(mean_eod)) * 1.0 / samplerate) - 0.5 * len(mean_eod) / samplerate) * 1000.0
+
+    if plot_data_func:
+        ax = plot_data_func(time, mean_eod, std_eod, **kwargs)
+        return mean_eod, std_eod, time, ax
+    else:
+        return mean_eod, std_eod, time
+
+if __name__ == '__main__':
+    try:
+        import config_tools as ct
+        import dataloader as dl
+        import bestwindow as bw
+        import psdtype as pt
+        import powerspectrum as ps
+    except ImportError:
+        'Import error !!!'
+        quit()
+
+    print('This modul detects single eods and tries to create a mean eod. Here this is shown as example for a defined file.')
+    print('')
```

Comment:
Well, I like it that way. Empty prints can be used to structure the output and the code.